src/clipping.jl

```diff
@@ -16,8 +16,36 @@ Clip the subject `geometry` with `other` geometry using clipping `method`.
 """
 function clip end
 
+function clip(poly::Polygon, other::Geometry, method::ClippingMethod)
+  b = boundary(other)
+  if b isa Multi
+    for r in parent(b)
+      poly = clip(poly, r, method)
+      if isnothing(poly)
+        return nothing
+      end
+    end
+    return poly
+  else
+    return clip(poly, b, method)
+  end
```

Comment:
This code pattern can be rewritten as two separate methods that dispatch on `Multi` or another geometry type:

```
_clip(poly, multi::Multi, method) = ...
_clip(poly, other, method) = ...
```

but I also think that we can delay this to a separate PR where we have more freedom to adjust the generic fallbacks that are unrelated to the specific clipping method.

Reply:
Applied the suggestion, function moved to weileratherton.jl. I am not sure this change is adding anything and even looks less clear to me. But I assume it can be the proper way to handle if further extensions are to be done.